Add redirects, sitemaps, and limits topics (#944)
commerce-docs/microsite-commerce-storefront: src/content/docs/setup/seo/platform-limits.mdx

import LinkCard from '@components/LinkCard.astro';
import { Aside } from '@astrojs/starlight/components';

Edge Delivery Services enforces limits on URL format, index size, sitemap file size, and redirect count.

## Document naming and URL format

URLs in Edge Delivery Services can contain only lowercase letters (`a-z`), numbers (`0-9`), and hyphens (`-`). Unsupported characters are replaced with hyphens, and leading or trailing hyphens are stripped. The full file path cannot exceed 900 characters.

For Commerce storefronts, this means:

- Product SKUs in URLs must be lowercase. The Commerce boilerplate enforces this automatically in the `getProductLink()` function.
- If you are migrating from a platform that used uppercase or special-character SKUs in URLs, add redirects from the old paths to the new lowercase ones. See [Redirects](/setup/seo/indexing/#redirects) for setup instructions.

If your existing URLs contain unsupported characters, add a CDN rewrite rule to normalize them before they reach the origin. See [Content delivery network](/setup/configuration/content-delivery-network/).

<Aside type="note" title="Lowercase SKU enforcement">

> Switching to lowercase SKUs on a live site changes URLs, which affects indexing. Plan the lowercase transition before launch.

Reviewer: ```suggestion
Switching to lowercase SKUs on a live site changes URLs, which affects indexing. Plan the transition to lowercase before launch.
```

</Aside>

## Indexing and sitemap size

> Index files have a maximum page count, and sitemap files have a maximum URL count and uncompressed size. See <Link href="https://www.aem.live/docs/limits#sitemap-limits" text="Sitemap limits on AEM.live" />.

Reviewer: ```suggestion
Index and sitemap files are subject to page count, URL count, and file size limits. See <Link href="https://www.aem.live/docs/limits#sitemap-limits" text="Sitemap limits on AEM.live" />.
```

> When a catalog is too large for a single index or sitemap file, you need multiple query indexes and matching sitemap files. To set this up, list all sitemap files in a `sitemap-index.xml` file and add each sitemap URL to `robots.txt`.

Reviewer: ```suggestion
For large catalogs, a single sitemap may not be sufficient. In that case, create multiple query indexes and corresponding sitemap files. Add all sitemap files to `sitemap-index.xml`, and reference each sitemap URL in `robots.txt`.
```

> If you use [AEM Commerce prerender](/setup/configuration/aem-prerender/) for a large catalog, plan the sitemap split before enabling prerender.

Reviewer: ```suggestion
If you use [AEM Commerce prerender](/setup/configuration/aem-prerender/) for a large catalog, plan your sitemap structure before enabling prerender.
```

See [Sitemaps](/setup/seo/sitemaps/) for configuration steps.

## Redirects

The redirect sheet has a maximum number of entries. If you have many URL changes, import redirects in phases. See <Link href="https://www.aem.live/docs/limits#redirect-limits" text="Redirect limits on AEM.live" /> for the limit.

See [SEO indexing — redirects](/setup/seo/indexing/#redirects) for Commerce-specific guidance.

## File upload limits

Each file type you upload through <Link href="https://da.live" text="Document Authoring" /> has a maximum size. These limits apply to content authors, not to storefront code or product data. Files that exceed the limit should be hosted on a CDN or third-party asset service and linked from your storefront pages.

For merchant guidance on when these limits apply, see [File and content limits](/merchants/edge-delivery-services/file-limits/).

> ## Full reference

Reviewer: ```suggestion
```
